### DISCUSSION TITLE
How to create a collection without vectors is field.

### DISCUSSION THREAD

**Discussion by pratikGhodke1** [milvus-io/milvus#29056] [Q&A and General discussion]
I want to create a collection for text matching and complementary data only. 

```python
CollectionSchema(
    [
        FieldSchema(
            name="id", is_primary=True, dtype=DataType.VARCHAR, max_length=30
        ),
        FieldSchema(name="name", dtype=DataType.VARCHAR, max_length=500),
    ],
    "Related data",
)
```

When I try to add the data I get the error - `pymilvus.exceptions.SchemaNotReadyException: <SchemaNotReadyException: (code=1, message=No vector field is found.)>`

When I add the vector field, the schema is generated without any issues but I have to provide the data for that field equal to the size other columns data and some vector dimension (x, y). If I pass an empty array or do not pass anything at all, I get - `<DataNotMatchException: (code=1, message=The data in the same column must be of the same type.)>, <Time:{'RPC start': '2023-12-07 16:27:18.775811', 'RPC error': '2023-12-07 16:27:18.775985'}>`.

How can I tackle this one?

Thanks

**Comment by yhmo** [comment]
Milvus is a vector database, each collection must has a vector field.
The python sdk example is a good place to learn how to create a collection and insert data: https://github.com/milvus-io/pymilvus/blob/master/examples/hello_milvus.py



Create a collection:

```
fields = [
    FieldSchema(name="id", is_primary=True, dtype=DataType.VARCHAR, max_length=30),
    FieldSchema(name="name", dtype=DataType.VARCHAR, max_length=500),
    FieldSchema(name="vector", dtype=DataType.FLOAT_VECTOR, dim=3)   # vector dimension is 3
]

schema = CollectionSchema(fields)
collection = Collection("TEST", schema)
```

Insert data according to the schema:
`collection.insert({"id": "ABC", "name": "Tom", "vector": [0.5, 1.0, 0.7]})`
